### PR TITLE
Fix N^2 behavior in roundstart logout report

### LINKED ISSUE #35715: Petsplosion triggers infinite loop detection in roundstart logout report
```
Infinite loop suspected--switching proc to background.
If it is not an infinite loop, either do 'set background=1' or set world.loop_checks=0.
proc name: display roundstart logout report (/proc/display_roundstart_logout_report)
  source file: game_mode.dm,397
  usr: (src)
  src: null
  call stack:
display roundstart logout report()
/datum/callback (/datum/callback): InvokeAsync()
Timer (/datum/controller/subsystem/timer): fire(0)
Timer (/datum/controller/subsystem/timer): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```

85c5a20e756ab7030af6c0b65fac843552bbfa82

### PR DESCRIPTION
**Pull request by SpaceManiac**
Fixes #35715.

### CODE REVIEW

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -412,7 +412,7 @@
 
 
 	for (var/C in GLOB.admins)
-		to_chat(C, msg)
+		to_chat(C, msg.Join(""))
```
Comment (lzimann) [inline code/game/gamemodes/game_mode.dm]: no need for the `""`

File: code/game/gamemodes/game_mode.dm [hunk 2]
```diff
@@ -370,7 +370,7 @@
 //Reports player logouts//
 //////////////////////////
 /proc/display_roundstart_logout_report()
-	var/msg = "<span class='boldnotice'>Roundstart logout report\n\n</span>"
+	var/list/msg = list("<span class='boldnotice'>Roundstart logout report\n\n</span>")
 	for(var/i in GLOB.mob_living_list)
 		var/mob/living/L = i
```
Comment (AnturK) [inline code/game/gamemodes/game_mode.dm]: You could probably also filter here for ever having a mind (last_mind)

Comment (Cyberboss) [reply code/game/gamemodes/game_mode.dm]: Additionally, you can reduce loop size by iterating mob_living_list & player_list

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Needful

**Comment by SpaceManiac** [comment]
Can't filter by `player_list` because the whole point is to catch mobs that aren't currently players.

**Comment by Cyberboss** [comment]
Player list is all mobs with a client I believe?

**Comment by SpaceManiac** [comment]
Right, and we need to check for mobs that have a `ckey` or a `last_mind` but *not* a `client`.

**Comment by AnturK** [comment]
Not wrong

**Pull request merged**